tensorflow/python/ops/array_ops.py

```diff
@@ -1388,13 +1390,6 @@ def stack(values, axis=0, name="stack"):
   Raises:
     ValueError: If `axis` is out of the range [-(R+1), R+1).
   """
-  if axis == 0:
-    try:
-      # If the input is a constant list, it can be converted to a constant op
-      return ops.convert_to_tensor(values, name=name)
-    except (TypeError, ValueError):
-      pass  # Input list contains non-constant tensors
-
```

Comment:
```
stacked = tf.stack(np.ones((2,0,3)))
print(stacked.shape)
# (2,0)
```
Quick fix.

Reply:
Why was this optimization removed?

Reply:
> Why was this optimization removed?

It breaks ```testZeroSizeCPU``` test case.

Reply:
What is the breakage? Does the test need to be fixed?

Reply:
> What is the breakage? Does the test need to be fixed?

stacked = tf.stack(list(np.ones((2,0,3))))
print(stacked.shape)
# (2,0)